OpenSSL 1.1 - SSLEAY_VERSION Not Declared In This Scope

```
...
  CXX      libbitcoin_server_a-init.o
In file included from addrman.h:13:0,
                 from init.cpp:12:
init.cpp: In function ‘bool AppInit2(boost::thread_group&, CScheduler&)’:
init.cpp:1029:60: error: ‘SSLEAY_VERSION’ was not declared in this scope
     LogPrintf("Using OpenSSL version %s\n", SSLeay_version(SSLEAY_VERSION));
                                                            ^
util.h:72:39: note: in definition of macro ‘LogPrintf’
 #define LogPrintf(...) LogPrint(NULL, __VA_ARGS__)
                                       ^
init.cpp:1029:74: error: ‘SSLeay_version’ was not declared in this scope
     LogPrintf("Using OpenSSL version %s\n", SSLeay_version(SSLEAY_VERSION));
                                                                          ^
util.h:72:39: note: in definition of macro ‘LogPrintf’
 #define LogPrintf(...) LogPrint(NULL, __VA_ARGS__)
                                       ^
At global scope:
cc1plus: warning: unrecognized command line option "-Wno-self-assign"
Makefile:3722: recipe for target 'libbitcoin_server_a-init.o' failed
make[2]: *** [libbitcoin_server_a-init.o] Error 1
```
Master (1.1): https://www.openssl.org/docs/manmaster/crypto/OPENSSL_VERSION_NUMBER.html
```
const char *OpenSSL_version(int t);
```
1.0.2: https://www.openssl.org/docs/man1.0.2/crypto/OPENSSL_VERSION_NUMBER.html
```
const char *SSLeay_version(int t);
```

--- a/src/init.cpp
+++ b/src/init.cpp
@@ -1026,7 +1026,12 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     if (fPrintToDebugLog)
         OpenDebugLog();
 
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
     LogPrintf("Using OpenSSL version %s\n", SSLeay_version(SSLEAY_VERSION));
+#else
+    LogPrintf("Using OpenSSL version %s\n", OpenSSL_version(OPENSSL_VERSION));
+#endif
+
 #ifdef ENABLE_WALLET
     LogPrintf("Using BerkeleyDB version %s\n", DbEnv::version(0, 0, 0));
 #endif

--- a/src/qt/rpcconsole.cpp
+++ b/src/qt/rpcconsole.cpp
@@ -263,7 +263,13 @@ RPCConsole::RPCConsole(const PlatformStyle *platformStyle, QWidget *parent) :
     connect(ui->btnClearTrafficGraph, SIGNAL(clicked()), ui->trafficGraph, SLOT(clear()));
 
     // set library version labels
+
+#if (OPENSSL_VERSION_NUMBER < 0x10100000L)
     ui->openSSLVersion->setText(SSLeay_version(SSLEAY_VERSION));
+#else
+    ui->openSSLVersion->setText(OpenSSL_version(OPENSSL_VERSION));
+#endif
+
 #ifdef ENABLE_WALLET
     ui->berkeleyDBVersion->setText(DbEnv::version(0, 0, 0));
 #else

